cf/data/dask_regrid.py

```diff
@@ -0,0 +1,590 @@
+            where ``I`` is the number of source grid cells and ``n``
+            is the number of regrid slices. Performance will be
+            optimised if *a* has Fortran order memory layout
+            (column-major order).
+
+        src_mask: `numpy.ndarray` or `None`
+            The source grid mask to be applied to the weights
+            matrix. If `None` then no source grid cells are masked. If
+            a Boolean `numpy` array then it must have shape ``(I,)``,
+            where ``I`` is the number of source grid cells, and `True`
+            signifies as masked cell.
+
+        weights: `numpy.ndarray`
+            The weights matrix that defines the regridding
+            operation. Might be modified (not in-place) to account for
+            missing data in *a*. Must have shape ``(J, I)``, where
+            ``J`` is the number of destination grid cells and ``I`` is
+            the number of source grid cells. Performance will be
+            optimised if *weights* has C order memory layout
+            (row-major order).
+
+            See `regrid` for details.
+
+        min_weight: float, optional
+            A very small non-negative number. It is used during linear
+            and first-order conservative regridding when adjusting the
+            weights matrix to account for the data mask. It is ignored
+            for all other regrid methods, or if data being regridded
+            has no missing values.
+
+            See `regrid` for details`
+
+        method: `str`
+            The name of the regridding method.
+
+        prev_mask: `numpy.ndarray` or `None`
+            The source grid mask used by a previous call to `_regrid`.
+            See *prev_weights* for details`.
+
+        prev_weights: `numpy.ndarray`, optional
+            The weights matrix used by a previous call to `_regrid`,
+            possibly modified to account for missing data. If
+            *prev_mask* equals *src_mask* then the *prev_weights*
+            weights matrix is used to calculate the regridded data,
+            bypassing any need to calculate a new weights matrix.
+            Ignored if `prev_mask` is `None`.
+
+    :Returns:
+
+        3-`tuple`
+            * `numpy.ndarray`: The regridded data.
+            * `numpy.ndarray` or `None`: The source grid mask applied
+              to the returned weights matrix, always identical to the
+              *src_mask* parameter.
+            * `numpy.ndarray`: The weights matrix used to regrid *a*.
+
+    """
+    if src_mask is None:
+        # ------------------------------------------------------------
+        # Source data is not masked
+        # ------------------------------------------------------------
+        w = weights
+    elif not src_mask.any():
+        # ------------------------------------------------------------
+        # Source data is not masked
+        # ------------------------------------------------------------
+        w = weights
```

Comment:
Could combine these first two statements with `or` to consolidate a bit?

Reply:
Yes: https://github.com/NCAS-CMS/cf-python/pull/438/commits/cb84bac6c30d68c70ab84eb8556b2636987df72a